### ISSUE TITLE
[nextgen-editor] Add an option to keep literal non-breaking space (U+00A0)

### ISSUE THREAD

**Issue by fdruide** [getgrav/grav-premium-issues#19]
Don’t worry, this is the last request I have in my notes. :-)

We noticed some pages containing non-breaking space (U+00A0) have been altered by normal ones when using nextgen-editor. 

This is indeed the case in markdown.js with one of the hookHTMLtoMarkdown  : 
```
    // replace non-breaking spaces with normal ones
    output = output.replace(/\u00A0/g, ' ');
```

This forces us to use the default editor because the behaviour is different. 

Is it possible to add an option to enable/disable this transformation? Actually, to keep the same behaviour as before, the default option might be to disable it. What do you think?

Tested with a clean install :
	•	Grav core + Admin (v1.7.0)
	•	NextGen Content Editor (v1.0.3)

Thank you!

**Comment by w00fz** [comment]
I can definitely make this an optional transformation so that the no-breaking spaces are preserved.

May I ask what is the context of this use? Are you pasting code that you have to leave untouched?
I am asking because It is generally considered bad practice to compensate spacing by adding `&nbsp;`, usually this should be dealt by padding, via CSS.

Cheers,

**Comment by fdruide** [comment]
We use non-breaking spaces purely for typographic purposes. It’s possible to directly type it (Option+Space on macOS) or it’s the job of a digital writing assistance tool. I have been working with linguists for 10 years and it is quite phenomenal how important it might be. I guess they debate on non-breaking vs. breaking while you and me, as developers, might debate on tab vs. space. :) 

While it is not really known to the general public, there are some specific rules to follow for publishers depending on the style guide.

Here is one example from The Chicago Manual of Style, 16th Edition :

> 6.11 Single quotation marks next to double quotation marks
> When single quotation marks nested within double quotation marks appear next to each other, no space need be added between the two except as a typographical nicety subject to the publisher’s requirements. For example, most typesetters will use a thin space between the two marks to enhance readability (as in the print edition of this manual). **In online works, a nonbreaking space should be used** (as in the online edition of this manual). See also 13.28. In the example that follows, note that the period precedes the single quotation mark (see also 6.9).
> 
> “Admit it,” she said. “You haven’t read ‘The Simple Art of Murder.’ ”

There are many more use cases in French, for example :

> 37,2_ºC
> x_>_y
> John_F. Kennedy
> Source : [https://www.druide.com/fr/enquetes/pour-des-espaces-insecables-impeccables](https://www.druide.com/fr/enquetes/pour-des-espaces-insecables-impeccables) (French only)

I hope my answer was informative.
Have a good day!

**Comment by w00fz** [comment]
More than informative, TIL, thank you :)

I will make sure to add that setting to avoid replacing non breaking spaces.

**Comment by w00fz** [comment]
This is now implemented in v1.0.5, it's a new entry in the Transformations options:

![image](https://user-images.githubusercontent.com/11734/105537221-760fb700-5ca6-11eb-864e-2dc5b6dcebbd.png)

**Comment by fdruide** [comment]
It works! Thank you, @w00fz :)
